larpandoradlcontent/LArTwoDReco/DLThreeDClusterSplittingAlgorithm.cc

```diff
@@ -0,0 +1,1350 @@
+                {
+                    int posIndex(splitPos.size());
+                    for (unsigned int i = 0; i < foundSplitPositions.size(); ++i)
+                    {
+                        splitPos.push_back(foundSplitPositions.at(i));
+                        clusterToSplitIndex[pCluster].push_back(posIndex);
+                        clusterFits.insert(std::make_pair(pCluster, clusterFit));
+                        ++posIndex;
+                    }
+                }
+            }
+            catch(...) { continue; }
+        }
+    }
+
+    // Leave if there's nothing to do
+    if (m_trainingMode || (splitPosU.empty() && splitPosV.empty() && splitPosW.empty()))
+        return STATUS_CODE_SUCCESS;
+
+    // Use 3D information to refine matches
+    IntVector usedU, usedV, usedW;
+    this->ThreeViewMatching(splitPosU, splitPosV, splitPosW, usedU, usedV, usedW);
+    this->TwoViewMatching(splitPosU, splitPosV, TPC_VIEW_U, TPC_VIEW_V, kdTreeW, usedU, usedV);
+    this->TwoViewMatching(splitPosU, splitPosW, TPC_VIEW_U, TPC_VIEW_W, kdTreeV, usedU, usedW);
+    this->TwoViewMatching(splitPosV, splitPosW, TPC_VIEW_V, TPC_VIEW_W, kdTreeU, usedV, usedW);
+
+    // Split the clusters
+    for (const HitType hitType : {TPC_VIEW_U, TPC_VIEW_V, TPC_VIEW_W})
+    {
+        // Adjust to view
+        PANDORA_RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=, 
+            PandoraContentApi::ReplaceCurrentList<Cluster>(*this, hitType == TPC_VIEW_U ? m_clusterListNameU :
+            hitType == TPC_VIEW_V ? m_clusterListNameV : m_clusterListNameW));
+        const IntVector &used(hitType == TPC_VIEW_U ? usedU : hitType == TPC_VIEW_V ? usedV : usedW);
+        CartesianPointVector &splitPos(hitType == TPC_VIEW_U ? splitPosU : hitType == TPC_VIEW_V ? splitPosV : splitPosW);
+        std::map<const Cluster*, IntVector> &clusterToSplitIndex(hitType == TPC_VIEW_U ? clusterToSplitIndexU :
+            hitType == TPC_VIEW_V ? clusterToSplitIndexV : clusterToSplitIndexW);
+        std::map<const Cluster*, TwoDSlidingFitResult> &clusterFits(hitType == TPC_VIEW_U ? clusterFitsU : 
+            hitType == TPC_VIEW_V ? clusterFitsV : clusterFitsW);
+
+        for (const auto& [pCluster, splitIndices] : clusterToSplitIndex) 
+        {
+            CartesianPointVector foundSplitPositions;
+
+            for (const int splitIndex : splitIndices)
+            {
+                if (std::find(used.begin(), used.end(), splitIndex) != used.end())
+                    foundSplitPositions.push_back(splitPos.at(splitIndex));
+            }
+
+            // Divide calo hits
+            if (clusterFits.find(pCluster) == clusterFits.end())
+                throw;
+
+            const TwoDSlidingFitResult &clusterFit(clusterFits.at(pCluster));
+            std::vector<CaloHitList> splitClusterHits(this->DivideCaloHits(clusterFit, foundSplitPositions));
+
+            if (splitClusterHits.size() < 2)
+                continue;
+
+            // Split clusters
```

Comment:
Redundant

Reply:
I felt that it was more appropriate to add a 

`if (foundSplitPositions.empty()) { continue; } `

further up to serve the purpose better